### DISCUSSION TITLE
ENUM numbers

### DISCUSSION THREAD

**Discussion by pcarruscag** [su2code/SU2#1220] [General]
@bigfooted @vdweide 
Line 5320 of CPhysicalGeometry.cpp:
`if(nSpanWiseSections[INFLOW -1] != nSpanWiseSections[OUTFLOW - 1]){`

We are indeed relying on at least one enum having specific values.

**Comment by vdweide** [comment]
Thanks @pcarruscag for finding out. Too bad this is the case though.

**Reply by bigfooted** [reply]
INFLOW=1, OUTFLOW=2. Do we really need an enum for this?

**Comment by pcarruscag** [comment]
Another good example of where things would break found by @TobiKattmann are the enums that start at 1 to then use a "NONE" case defined somewhere else, the NONE (0) case is often implicitly converted to boolean.
So I think the only safe way to go about this is to replace normal `enum` by `enum class` as Tobi started doing in #1242.
Off course, for those where the numbers already start at zero it would also be safe to delete them.
